### PR TITLE
Launch file for combined gzserver + bridge

### PR DESCRIPTION
**Pull request by caguero**
# 🎉 New feature

Part of https://github.com/gazebosim/ros_gz/issues/544

## Summary

This patch lets you combine the functionality from #530 and #532 into a single launch file.

### How to test it?

Use the provided `ros_gz_sim.launch.py` (modify `config_file` accordingly) to run the bridge as an executable

```
ros2 launch ros_gz_sim ros_gz_sim.launch.py world_sdf_file:=empty.sdf config_file:=/home/caguero/ros_gz_ws/src/ros_gz/ros_gz_bridge/test/config/full.yaml
```

You should see messages of gzserver loading. Additionally, verify that you see the `/ros_chatter` topic:
```
caguero@cold:~/ros_gz_ws$ ros2 topic list
/parameter_events
/ros_chatter
/rosout
```

Confirm that no container has been created:
```
caguero@cold:~/ros_gz_ws$ ros2 component list
```

Now `CTRL-C` to stop all nodes and try the composable version:
```
ros2 launch ros_gz_sim ros_gz_sim.launch.py world_sdf_file:=empty.sdf use_composition:=True config_file:=/home/caguero/ros_gz_ws/src/ros_gz/ros_gz_bridge/test/config/full.yaml
```

gzserver should be running. Verify that you also see the `/ros_chatter` topic:
```
caguero@cold:~/ros_gz_ws$ ros2 topic list
/parameter_events
/ros_chatter
/rosout
```

And confirm that a container has been created with both nodes:
```
caguero@cold:~/ros_gz_ws$ ros2 component list
/ros_gz_container
  1  /ros_gz_bridge
  2  /gzserver
```

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: ros_gz_sim/launch/ros_gz_sim.launch.py [hunk 1]
```diff
@@ -13,59 +13,100 @@
...
+        PythonLaunchDescriptionSource(
+            [PathJoinSubstitution([FindPackageShare('ros_gz_bridge'),
+                                   'launch',
+                                   'ros_gz_bridge.launch.py'])]),
+        launch_arguments=[('config_file', config_file),
+                          ('container_name', container_name),
+                          ('namespace', namespace),
+                          ('use_composition', use_composition),
+                          ('use_respawn', use_respawn),
+                          ('log_level', log_level), ])
```
Comment (ahcorde) [inline ros_gz_sim/launch/ros_gz_sim.launch.py]:
```suggestion
                          ('log_level', log_level)])
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: ros_gz_sim/launch/ros_gz_sim.launch.py [hunk 2]
```diff
@@ -13,59 +13,100 @@
...
 
 def generate_launch_description():
 
-    world_sdf_file_arg = DeclareLaunchArgument(
+    config_file = LaunchConfiguration('config_file')
+    container_name = LaunchConfiguration('container_name')
+    namespace = LaunchConfiguration('namespace')
+    use_composition = LaunchConfiguration('use_composition')
+    use_respawn = LaunchConfiguration('use_respawn')
+    log_level = LaunchConfiguration('log_level')
```
Comment (azeey) [inline ros_gz_sim/launch/ros_gz_sim.launch.py]: I imagine we'll have a `log_level` parameter for gz_sever as well at some point. How would we distinguish parameters for the server and parameters for the bridge? Should we rename this to `bridge_log_level`?

File: ros_gz_sim/launch/ros_gz_sim.launch.py [hunk 3]
```diff
@@ -13,59 +13,100 @@
...
+from launch.actions import DeclareLaunchArgument, IncludeLaunchDescription
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch.substitutions import LaunchConfiguration, PathJoinSubstitution, TextSubstitution
+from launch_ros.substitutions import FindPackageShare
 
 
 def generate_launch_description():
 
-    world_sdf_file_arg = DeclareLaunchArgument(
+    config_file = LaunchConfiguration('config_file')
```
Comment (azeey) [inline ros_gz_sim/launch/ros_gz_sim.launch.py:27]: Any luck with refactoring these into a separate Python module so we don't have to duplicate code?

**Review by ahcorde — APPROVED** [review]

File: ros_gz_sim/launch/ros_gz_sim.launch.py (hunk 3 above)
Comment (caguero) [reply ros_gz_sim/launch/ros_gz_sim.launch.py:27]: > Any luck with refactoring these into a separate Python module so we don't have to duplicate code?

Apparently this is possible by creating a separate launch file for the common parts. I'll do it in a separate PR.

**Comment by caguero** [comment]
@ahcorde, is CI supposed to pass right now with 24.04?

**Comment by ahcorde** [comment]
@caguero  yes, I think it should pass

File: ros_gz_sim/launch/ros_gz_sim.launch.py (hunk 2 above)
Comment (caguero) [reply ros_gz_sim/launch/ros_gz_sim.launch.py]: Renamed in [66e74f8](https://github.com/gazebosim/ros_gz/pull/533/commits/66e74f8fbe11cdd2412ba36d79ad827d5182190c)

**Pull request merged**